cpp/src/lib/cucumber-messages/CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+add_library(cucumber-messages_library)
+add_library(cucumber::messages ALIAS cucumber-messages_library)
+
+set(INC_DIR "${CMAKE_SOURCE_DIR}/include")
+
+target_sources(
+    cucumber-messages_library
+    PRIVATE
+        ${INC_DIR}/cucumber/messages/attachment_content_encoding.hpp
```

Comment:
It might be prudent to generate this list as well. Otherwise adding a new message may break the CPP project.

Reply:
Agreed. While it's considered an "anti-pattern" in CMake, I've always preferred it that way.

Changed accordingly.

As for the C++20 dependency, it requires a fairly recent compiler and may fail in some environments. Only a large and diverse user base could tell if this is a problem (enhanced language constructs, features and standard library).
It's reasonable to expect C++20 nowadays, but I cannot speak for other human beings...

Reply:
You're currently the only active C++ contributor.  I think we can use you as representative sample of the userbase. 😄